### PR TITLE
Adds in reducers for tracking min/max and standard deviation.

### PR DESCRIPTION
**Pull request by jtrobec**
Adding in two reducers to be used with the ReducedMetric, for tracking min/max and standard deviation. I used the MeanReducer as a starting point, hopefully these follow the format you're interested in using.

### CODE REVIEW

File: src/jvm/backtype/storm/metric/api/MinMaxReducer.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  @Override
+  public MinMaxReducerState init() {
+    return new MinMaxReducerState();
+  }
+
+  @Override
+  public MinMaxReducerState reduce(MinMaxReducerState acc, Object input) {
+    double value;
+    if(input instanceof Double) {
```
Comment (nathanmarz) [inline src/jvm/backtype/storm/metric/api/MinMaxReducer.java:51]: Simpler to do ((Number) input).doubleValue()

File: src/jvm/backtype/storm/metric/api/StdDevReducer.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+public class StdDevReducer implements IReducer<StdDevReducerState> {
+  public StdDevReducerState init() {
+    return new StdDevReducerState();
+  }
+
+  public StdDevReducerState reduce(StdDevReducerState acc, Object input) {
+    acc.count++;
+    double value;
+    if(input instanceof Double) {
```
Comment (nathanmarz) [inline src/jvm/backtype/storm/metric/api/StdDevReducer.java:19]: Simpler to do ((Number) input).doubleValue()

**Comment by nathanmarz** [comment]
Made a couple of inline comments. Can you also make the pull request against master instead of 0.9.0?

**Comment by jtrobec** [comment]
Sure, will do -- will also go ahead and update the existing MeanReducer to
use the ((Number) value).doubleValue() pattern...will resubmit in a day or
two.

**Comment by nathanmarz** [comment]
Closing as these changes were never made. Feel free to open a new pull request with the appropriate changes, as would like to pull this in.

**Pull request closed without merging**